src/google-analytics.js

```diff
@@ -0,0 +1,219 @@
+const PCT_75_ACTION: string = '75 pct watched';
+const PCT_100_ACTION: string = '100 pct watched';
+
+/**
+ * Your class description.
+ * @classdesc
+ */
+export default class GoogleAnalytics extends BasePlugin {
+  /**
+   * The default configuration of the plugin.
+   * @type {Object}
+   * @static
+   */
+  static defaultConfig: Object = {
+    tracking: defaultTracking
+  };
+
+  /**
+   * @static
+   * @public
+   * @returns {boolean} - Whether the plugin is valid.
+   */
+  static isValid(): boolean {
+    return true;
+  }
+
+  /**
+   * The gtag lib url.
+   * @type {string}
+   * @static
+   */
+  static GTAG_LIB_URL: string = "//www.googletagmanager.com/gtag/js";
+
+  /**
+   * Indicate whether time percent event already sent
+   * @private
+   */
+  _timePercentEvent: { [event: string]: boolean } = {
+    PLAY_REACHED_25: false,
+    PLAY_REACHED_50: false,
+    PLAY_REACHED_75: false,
+    PLAY_REACHED_100: false
+  };
+
+  /**
+   * @constructor
+   * @param {string} name - The plugin name.
+   * @param {Player} player - The player instance.
+   * @param {Object} config - The plugin config.
+   */
+  constructor(name: string, player: Player, config: Object) {
+    super(name, player, config);
+    this._init();
+    this._addBindings();
+    this._sendEvent({
+      action: WIDGET_LOADED_ACTION,
+      category: this.config.tracking.category
+    });
+  }
+
+  /**
+   * _gtag - update the dataLayer for gtag manager
+   * @private
+   * @returns {void}
+   */
+  _gtag() {
+    window.dataLayer.push(arguments);
+  }
+
+  /**
+   * _init
+   * @private
+   * @returns {void}
+   */
+  _init(): void {
+    if (this.config.trackingId) {
+      if (!window.google_tag_manager) {
+        Utils.Dom.loadScriptAsync(`${GoogleAnalytics.GTAG_LIB_URL}?id=${this.config.trackingId}`).then(() => {
+          this.logger.debug('Google gtag library has loaded successfully');
+        });
+      }
+      window.dataLayer = window.dataLayer || [];
+      // $FlowFixMe
+      this._gtag('js', new Date());
+      // $FlowFixMe
+      this._gtag('config', this.config.trackingId);
+    }
+  }
+
+  /**
+   * _addBindings
+   * @private
+   * @returns {void}
+   */
+  _addBindings(): void {
+    const shouldSentEvent = (condition, event) => {
+      return typeof condition === 'function' ? condition.call(this, event) : true;
+    };
+
+    Object.entries(this.config.tracking.events).forEach(([eventName, eventParams]) => {
+      this.eventManager.listen(this.player, this.player.Event[eventName], (event) => {
+        try {
+          if (eventParams && typeof eventParams === 'object' && eventParams.action && shouldSentEvent(eventParams.condition, event)) {
+            const eventObj = {
+              action: eventParams.action,
+              category: eventParams.category || this.config.tracking.category,
+              label: typeof eventParams.label === 'function' ? eventParams.label.call(this, event) : this.config.tracking.label.call(this, event),
```

Comment:
Let's make this behaviour more generic for all event keys.
Check if event value exist and if it's function or string, otherwise take it from default config